### DISCUSSION TITLE
unable to authenticate with generated ssh key

### DISCUSSION THREAD

**Discussion by imperator08** [community/community#127273] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi all,

I need support regarding ssh authentication. I have created a new ssh key on my mac book based on the following tuorials:

https://docs.github.com/de/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent

https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account?tool=webui

the key is also listed in my account:
<img width="942" alt="Bildschirmfoto 2024-06-04 um 21 09 35" src="https://github.com/community/community/assets/171152052/c998ae1f-9c64-49eb-b5b5-f0d5af6838f4">

Nevertheless I am asked to authenticate when I run the following command in my terminal:

git push -u origin main  

After I have inserted my username and password correctly, I am receiving the following message:

remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for

Is there anything what I missed to configure?

**Comment by davevad93** [comment]
Hi @imperator08 ,

That's strange it shouldn't happen when you `git clone`, `git fetch`, `git pull`, or `git push` to a remote repository using `SSH` URLs. As the error states `Support for password authentication was removed on August 13, 2021` but that's just for `HTTPS` URLs as password auth has been replaced by `personal access token` for security reasons, here's how to create one: [Managing your personal access tokens](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens). When you have generated the token now you can use it as a password for your `git` operations but as I said before, it's necessary only with `HTTPS` as stated in the [docs](https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls). I always clone and push to my remote repos using `SSH` and I can use my user pass without any problem.

**Reply by davevad93** [reply]
Maybe there's some git configuration issue, make sure your email is the same as the one associated to your GitHub account by running: 
`git config --global user.email your@email.com`
